pkg/ctl/functions/create.go

```diff
@@ -420,21 +424,16 @@ func doCreateFunctions(vc *cmdutils.VerbCmd, funcData *util.FunctionData) error
 
 	admin := cmdutils.NewPulsarClientWithAPIVersion(common.V3)
 
-	if utils.IsPackageURLSupported(funcData.Jar) {
-		err = admin.Functions().CreateFuncWithURL(funcData.FuncConf, funcData.Jar)
-		if err != nil {
-			cmdutils.PrintError(vc.Command.OutOrStderr(), err)
-		} else {
-			vc.Command.Printf("Created %s successfully\n", funcData.FuncName)
-		}
+	if utils.IsPackageURLSupported(funcData.UserCodeFile) {
```

Comment:
From examining the code `funcData.UserCodeFile` should suffice here, since:

1. `UserCodeFile` will have the value provided as a location to any one of the `--jar` `--py` and `--go` options already
2. All three types should now support URL deployment
3. the check I added in `util.go` should ensure that only one of the flags will set this value